src/main/java/org/openstreetmap/atlas/generator/tools/streaming/ResourceFileSystem.java

```diff
@@ -184,7 +196,21 @@ public boolean rename(final Path source, final Path destination) throws IOExcept
             // Hadoop end of job rename case
             destinationName = destinationName.endsWith("/") ? destinationName
                     : destinationName + "/";
-            final String appendName = sourceName.substring(sourceName.lastIndexOf("/") + 1);
+            // Sample resource:
+            // resource://test/atlas/_temporary/0/_temporary/attempt_201804061442_0003_m_000002_14/DMA/9/DMA_9-168-234.atlas
+            final Pattern pattern = Pattern.compile("\\/attempt_[^\\/]*\\/");
+            final Matcher matcher = pattern.matcher(sourceName);
+            final String appendName;
+            if (matcher.find())
+            {
+                final String match = matcher.group(0);
+                final StringList split = StringList.split(sourceName, match);
+                appendName = split.get(1);
```

Comment:
Is there any chance the folder could be empty, and .get(1) could return IndexOutOfBounds?

Reply:
This `ResourceFileSystem` is used for tests mostly. Here it is doing a big hack to conform to the Spark file naming, and yes if they change it it will break. Not sure there is a better way right now..